Implement Start Tracking Functionality

dhargrow opened the pull request:
**Overview**

This PR  establishes a new protocol that defines time tracking functionality for tasks, currently there is one static func startTracking that returns a Date object. Also a monostate singleton was added to conform to the new protocol which has a static property that should be the single source of truth for tasks 

**Key Changes**

1. `TimeTracking` Protocol:
    - `Session` structure has been added to represent individual time tracking sessions.
2. `SessionManager` Struct
    - Conforms to TimeTracking and implements its methods
    - Contains static property:
      - `sessions`: A dictionary of key type string and value type `Session`
3. Unit Tests for `SessionManager`
    - Added comprehensive tests in `SessionManagerTests.swift` that cover various scenarios, including:
       - Return a Date when a session exists in `SessionManager.sessions`
       - Return a Date when a session doesn't exists in `SessionManager.session`

dhargrow committed:
Creating protocol that defines time tracking functionality for tasks

dhargrow committed:
Making func static

dhargrow committed:
Creating monostate singleton to conform to TimeTracking protocol, along with tests

Tavernari commented on Sources/SwiftCuckoo/SessionManager.swift:
```diff
@@ -0,0 +1,24 @@
+    private init() {}
+
+    /// stores all sessions.
+    static public var sessions: [String: Session] = [:]
```
It is not safe to be used in a concurrency environment.

Tavernari committed:
Refactor to accept concurrency and make it more testable

Tavernari reviewed (changes requested):
Before continuing:
 
I opened a PR pointing to this one, suggesting changes to make it more testable and ready for Swift Concurrency features like Actors.
 
Let me know your thoughts on it.
 
https://github.com/Tavernari/SwiftCuckoo/pull/20

Tavernari committed:
Change protocol conformance from AnyActor to Sendable

Replaced `AnyActor` with `Sendable` in the `SessionManagerStorage` protocol to align with concurrency or safety requirements.

Tavernari committed:
Fix typo in README.md

Corrected a typo in the "Change Notifications" section by replacing "ppply" with "apply."

Tavernari committed:
Add platform requirements section to README.md

Included a new "Platforms" section in the README.md specifying minimum platform versions for macOS (10.15+) and iOS (13.0+).

Tavernari committed:
Merge pull request #20 from Tavernari/issue3-start-tracking-refactor

Refactor to accept concurrency and make it more testable
